staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -381,7 +381,7 @@ func (az *Cloud) serviceOwnsFrontendIP(fip network.FrontendIPConfiguration, serv
 			return false, isPrimaryService, nil
 		}
 
-		return false, isPrimaryService, fmt.Errorf("serviceOwnsFrontendIP: wrong parameters")
+		return false, isPrimaryService, nil
```

Comment:
As it stands the caller cannot tell the difference between this returned response and the response a few lines earlier (ie. whether or not the pip/fip were fully configured. ) We could just lose the earlier return if we really don't care about differentiating. Also are we sure that we might not hit a problem where something after calling serviceOwnsFrontendIP() references PublicIPAddress safely because currently this ensures its not nil?

Reply:
yes, that's a good suggestion. We could improve in the out-of-tree repo. And we're sure that the nil properties won't cause issues after the check (the nil pointer would be rechecked again whenever it is consumed again).